faculty/clients/api.py

```diff
@@ -0,0 +1,635 @@
+        project_id : uuid.UUID
+            The project in which the API resides.
+        api_id : uuid.UUID
+            The API in which to the production key resides.
+        key_id : uuid.UUID
+            The key to disable.
+        """
+        return self._toggle_production_key(project_id, api_id, key_id, False)
+
+    def update_production_key(self, project_id, api_id, key_id, label):
+        """Disable a specific production API key.
+
+        Parameters
+        ----------
+        project_id : uuid.UUID
+            The project in which the API resides.
+        api_id : uuid.UUID
+            The API in which to the production key resides.
+        key_id : uuid.UUID
+            The key to update.
+        label : str
+            The new label to set to the given key.
+
+        Returns
+        -------
+        APIKey
+            The updated key.
+        """
+        endpoint = "/project/{}/api/{}/key/{}".format(
+            project_id, api_id, key_id
+        )
+        payload = {"label": label}
+        return self._put(endpoint, _APIKeySchema(), json=payload)
+
+    def delete_production_key(self, project_id, api_id, key_id):
+        endpoint = "/project/{}/api/{}/key/{}".format(
+            project_id, api_id, key_id
+        )
+        self._delete_raw(endpoint)
+
+    def start(
+        self,
+        project_id,
+        api_id,
+        key_id,
+        server_type=None,
+        image_version=None,
+        current_api=None,
+    ):
+        if not current_api:
+            current_api = self._get_current_api_definition(project_id, api_id)
+        payload = {
+            "instanceSizeType": current_api["instanceSizeType"],
+            "instanceSize": {
+                "milliCpus": current_api["instnaceSize"]["milliCpus"],
+                "memoryMb": current_api["instnaceSize"]["memoryMb"],
+            },
+        }
+        if server_type:
+            payload["instanceSizeType"] = server_type["instance_size_type"]
+            if server_type["instance_size"]:
+                instance_size = server_type["instance_size"]
+                payload["instanceSize"] = {
+                    "milliCpus": instance_size.milli_cpus,
+                    "memoryMb": instance_size.memory_mb,
+                }
+        if image_version:
+            payload["imageVersion"] = image_version
+        endpoint = "/project/{}/api/{}/prod/start".format(project_id, api_id)
+        return self._put(endpoint, APIInstanceResponseSchema(), json=payload)
+
+    def stop(self, project_id, api_id):
+        endpoint = "/project/{}/api/{}/prod/stop".format(project_id, api_id)
+        self._put(endpoint, APIInstanceResponseSchema())
+
+    def restart(
+        self,
+        project_id,
+        api_id,
+        key_id,
+        server_type=None,
+        image_version=None,
+        current_api=None,
+    ):
+        if not current_api:
+            current_api = self._get_current_api_definition(project_id, api_id)
```

Comment:
As above

Reply:
I just side-stepped this by not loading the API, though it might be a little less ergonomic even if simpler internally.